apis/duck/v1/kresource_type.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"knative.dev/pkg/apis"
+)
+
+// DuckShaped is an interface for retrieving the duck elements of an arbitraty resource.
```

Comment:
total bikeshed, but curious about such broad DuckShaped vs. something in the name about it being for ducks with status? Not all the duck definitions have Status. Shrug, thoughts? For example, would KResourceShaped be a better description?

Reply:
I did not know we had non-status containing ducks. KRShaped is fine too. I'm open to any naming suggestions.